LiteCore/Query/QueryParser.cc

```diff
@@ -352,11 +352,50 @@ namespace litecore {
         return count;
     }
 
+namespace {
+    bool canDefaultTableHaveDeleted() { return true; }
```

Comment:
Why function instead of a constant value? Do you plan to have some logic implemented here in the future?

Reply:
I am envisioning in the future, this will be a function for DataFile::Delegate. When in open, it found no deleted docs in normal keystore, the function returns false and effectively disable all the new code. Of course, all affected test cases will have to be adjusted.

Reply:
I see, you want to use this to check if the deleted docs in the kv_default are all moved in the future.